### PR TITLE
Add timeouts setup from environment

### LINKED ISSUE #258: Implement ability to get test-run timeouts from the hosts local configurations
Trying to fix #251 found that on old hosts vinyl/ suite tests failed to finish in default timeout '--test-timeout' == 110 seconds. Also it happened because of slow IO with slow hard drives setup on hosts. In major it happened on box.snapshot() calls, but could happen on any other commands.
To fix it was decided to collect statistics and setup on hosts local timeout configurations.
Found that running in complete parallel mode for all the tests even from fragile lists then test-timeout value had results:
  110 (default) -> a lot of fails
  210 -> fails rare, but still exists in different places, a lot of checksums needed to pass
  310 -> fails very rare, exists in a few places only, small number of checksums may help to pass it

### PR DESCRIPTION
**Pull request by avtikhon**
Added ability to set using environment the following timeouts based on
appropriate run options:
```
  TEST_TIMEOUT			--test-timeout
  NO_OUTPUT_TIMEOUT		--no-output-timeout
  REPLICATION_SYNC_TIMEOUT	--replication-sync-timeout
```
Closes #258

### CODE REVIEW

**Review by okoshovetc — APPROVED** [review]
LGTM

File: dispatcher.py [hunk 1]
```diff
@@ -143,10 +143,9 @@ def init_listeners(self):
             self.listeners.append(self.fail_watcher)
         if watch_hang:
             warn_timeout = 60.0 if args.long else 10.0
-            no_output_timeout = float(args.no_output_timeout or 120)
             hang_watcher = listeners.HangWatcher(
                 output_watcher.not_done_worker_ids, self.kill_all_workers,
-                warn_timeout, no_output_timeout)
+                warn_timeout, args.no_output_timeout)
```
Comment (Totktonada) [inline dispatcher.py:148]: It seems, it does not break anything, but I would keep the `float` type. At least, it is what HangWatcher expects.

File: lib/options.py [hunk 2]
```diff
@@ -193,15 +193,15 @@ def __init__(self):
         parser.add_argument(
                 "--test-timeout",
                 dest="test_timeout",
-                default=110,
+                default=env_int('TEST_TIMEOUT', 110),
```
Comment (Totktonada) [inline lib/options.py:196]: Since those values may be passed implicitly now, I would show them on the terminal before start testing.

**Review by Totktonada — APPROVED** [review]
LGTM aside of two non-critical comments.

File: lib/options.py (hunk 2 above)
Comment (avtikhon) [reply lib/options.py:196]: Sure, added print block, looks like:
```
Running in parallel with 16 workers

Timeout variables:
-------------------
REPLICATION_SYNC_TIMEOUT: 100
TEST_TIMEOUT:             110
NO_OUTPUT_TIMEOUT:        120

Collecting tests in 'app'            (Found 0   tests): application server tests.
```

File: dispatcher.py (hunk 1 above)
Comment (avtikhon) [reply dispatcher.py]: Ok, sure, added float().

**Pull request merged**